.github/workflows/maven.yml

```diff
@@ -33,20 +33,11 @@ jobs:
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml', '**/*.target') }}
           restore-keys: ${{ runner.os }}-m2
 
-      - name: Show Maven version
-        run: |
-          which mvn
-          mvn -version
-
-      - name: Build and test (headless with xvfb in Linux)
-        run: xvfb-run mvn clean verify -PuseJenkinsSnapshots -Dsurefire.rerunFailingTestsCount=3
-        if: matrix.os == 'ubuntu-latest'
-        working-directory: org.eclipse.xtext.full.releng
-
-      - name: Build and test (in other OSes)
-        run: mvn clean verify -PuseJenkinsSnapshots -Dsurefire.rerunFailingTestsCount=3
-        if: matrix.os != 'ubuntu-latest'
-        working-directory: org.eclipse.xtext.full.releng
+      - name: Build and test
```

Comment:
question would be: why is xvfb-run insufficient.
we used that cause the original GabrielBB/xvfb-action
was no longer maintained
coactions/setup-xvfb
seems to be the new fork.
but i am not sure if needed cause xvfb-run seem to work.
should this be reported to coactions/setup-xvfb

Reply:
> Besides not having the "if" and 2 steps, does the use of this action provide additional benefits?

No more benefits I'm aware of.

> Have you used this action in other projects?

Yes, for example we used the predecessor (as Christian mentioned) in M2E for a while and I just changed to the new one recently: https://github.com/eclipse-m2e/m2e-core/pull/1344
And SWT uses it as well (I should probably update it there too):
https://github.com/eclipse-platform/eclipse.platform.swt/blob/6912a16abe78c905180908ea4d59218f6519c272/.github/workflows/maven.yml#L60

> question would be: why is xvfb-run insufficient.

Probably, yes. I have not digged into the details, but as far as I understand this is mainly for conveniences and avoids the need for the if.
From https://github.com/coactions/setup-xvfb

`This action installs [XVFB](http://elementalselenium.com/tips/38-headless) and runs your headless tests with it. It cleans up the xvfb process after your tests are done. If it detects you're not using linux then your tests still run, but without xvfb, which is very practical for multi-platform workflows.`

So that's probably it.

Reply:
we were using the garielbb for rerference project integ tests nor about 5 years
but we stopped using once it was no longer maintained
and github was issuing warnings about the deprecation feature use of that
this is why we then switchted to the alternative approach
https://github.com/itemis/xtext-reference-projects/commit/bc0256f440b2b03c9527fa466883f04b53f09cd6

Reply:
> this is why we then switchted to the alternative approach

Understand, but since the action has been revived I think things changed a bit.
Any way, just let me know which way you want and I can act accordingly.

Reply:
using the new action is ok. the question is the warning already reported upstream or do we do something wrong

Reply:
OK. 👍🏽 
Probably not reported, at least I cannot see any issue on that topic.
This extra package installation was just recently added in https://github.com/coactions/setup-xvfb/pull/5. Unfortunately there is not much context.
Maybe this can be made smarter by just installing the recommended packages with it, but I'm not an apt expert?

Reply:
hmmm ok

Reply:
I don't see the warning in GitHub Actions.
If the action has been revived, I'm OK with that.
The apt installation doesn't seem to take a lot of time.
If something goes wrong we can revert.
@cdietrich I'd approve the PR, what about you?